packages/pyright-internal/src/analyzer/analyzerNodeInfo.ts

```diff
@@ -81,108 +81,110 @@ export type ScopedNode = ModuleNode | ClassNode | FunctionNode | LambdaNode | Co
 // Cleans out all fields that are added by the analyzer phases
 // (after the post-parse walker).
 export function cleanNodeAnalysisInfo(node: ParseNode) {
-    const analyzerNode = node as AnalyzerNodeInfo;
-    delete analyzerNode.scope;
-    delete analyzerNode.declaration;
-    delete analyzerNode.flowNode;
-    delete analyzerNode.afterFlowNode;
-    delete analyzerNode.fileInfo;
-    delete analyzerNode.codeFlowExpressions;
-    delete analyzerNode.codeFlowComplexity;
-    delete analyzerNode.dunderAllInfo;
-    delete analyzerNode.typeParameterSymbol;
+    const analyzerNode = getAnalyzerInfo(node);
+    if (analyzerNode) {
+        delete analyzerNode.scope;
+        delete analyzerNode.declaration;
+        delete analyzerNode.flowNode;
+        delete analyzerNode.afterFlowNode;
+        delete analyzerNode.fileInfo;
+        delete analyzerNode.codeFlowExpressions;
+        delete analyzerNode.codeFlowComplexity;
+        delete analyzerNode.dunderAllInfo;
+        delete analyzerNode.typeParameterSymbol;
```

Comment:
This categorically deoptimizes the internal maps. If possible set them to `undefined` instead.